adapters/tlsio_openssl.c

```diff
@@ -1022,7 +1022,8 @@ static int enable_domain_check(TLS_IO_INSTANCE* tlsInstance)
         X509_VERIFY_PARAM *param = SSL_get0_param(tlsInstance->ssl);
 
         X509_VERIFY_PARAM_set_hostflags(param, 0);
-        if (!X509_VERIFY_PARAM_set1_host(param, tlsInstance->hostname, strlen(tlsInstance->hostname)))
+        if (!(X509_VERIFY_PARAM_set1_ip_asc(param, tlsInstance->hostname) ||
```

Comment:
>X509_VERIFY_PARAM_set1_ip_asc [](start = 14, length = 29)

can we do host first and then ip. Because most people will use hostname ?

Reply:
`set1_host` will always work, even when an ASCII IP Address is used within the `hostname` variable. 
We don't want that because OpenSSL, while not allowed by RFC, will check that ASCII-encoded IPv4/v6 address within a DNS Name-typed SAN entry, something we don't want.

The code will only check IP Address-typed SAN entries when IPs are entered. If we fail to parse the IP (first function will fail) then we fallback to consider the hostname a non-IP entity and the process described here: https://wiki.openssl.org/index.php/Hostname_validation